### ISSUE TITLE
no fmt for HTTPRequestRule

### ISSUE THREAD

**Issue by LeoK80** [haproxytech/dataplaneapi#348]
**TL;DR**
I'm trying to add a simple path rewrite rule to drop a specific first part of a path when received from the front-end. I'm not sure if I'm overlooking something in the api docs or whether I'm dealing with some bugs here. I really would appreciate any feedback.

**Method 1 - set-uri**
Currently using haproxy v2 and dataplane api and running into and issue trying to rewrite the path.
Goal is to add a rule to a backend in haproxy:
```
http-request set-uri %[url,regsub(^/metrics/,/,)] if { path_beg /metrics }
```
I've been using the url as per documentation with the needed query params:
```
/v2/services/haproxy/configuration/http_request_rules?transaction_id={{transactionid}}&parent_name={{backend_name}}&parent_type=backend
```
With the JSON payload on POST request:
```
{
    "type": "set-uri",
    "index": 0,
    "expr": "%[url,regsub(^/metrics/,/,)]",
    "cond": "if",
    "cond_test": "{ path_beg /metrics }"
}
```
I've also tried with the `"uri-match"` and `"uri-fmt"` as well as `"path_match"` and `"path_fmt"`, however the result is always the same. It ends up with an empty space for the expression used (mind, **two spaces** instead of one!):
```
http-request set-uri  if { path_beg /metrics }
```
**The issue**:
I can't seem to get the required expression into Haproxy using the dataplane api for HTTPRequestRule

**Method 2 - path-replace**
The other attempted approach is using the `path-replace` approach.

I've been using the url as per documentation with the needed query params:
```
/v2/services/haproxy/configuration/http_request_rules?transaction_id={{transactionid}}&parent_name={{backend_name}}&parent_type=backend
```
With the JSON payload on POST request:
```
{
    "type": "replace-path",
    "index": 0,
    "cond": "if",
    "cond_test": "{ url_beg /metric/ }",
    "path_fmt": "/\\1",
    "path_match": "/metrics/(.*)"
}
```
The resulting rule in `haproxy.cfg`:
```
http-request replace-path /metrics/(.*) /\\1 if { url_beg /metrics/ }
```

**The issue: escape carries into the haproxy rules**
It seems that the escape backslash used to satisfy JSON standards, carries on into the Haproxy config rule instead of being dropped. I'd expect this to break the look-back reference in `haproxy.cfg`

Expected output into `haproxy.cfg`:
```
http-request replace-path /metrics/(.*) /\1 if { url_beg /metrics/ }
```

**Comment by mjuraga** [comment]
Hi thanks for reporting this, we will take a look and get back to you.

**Comment by oliwer** [comment]
Hi @LeoK80. Currently, to use "set-uri" you must set "hdr_format" instead of "expr". But this is a bug. As shown in [the spec](https://github.com/haproxytech/client-native/blob/master/specification/models/configuration/http/request.yaml#L308), it should be "uri-fmt". I will fix this for the next version.

**Comment by LeoK80** [comment]
thanks @oliwer I will give it a try with that and see where I get at!
